aws/ec2/ec2/vm.go

```diff
@@ -89,6 +83,25 @@ func NewEC2Instance(e aws.Environment, name, ami, arch, instanceType, keyPair, u
 		Tags: pulumi.StringMap{
 			"Name": e.Namer.DisplayName(pulumi.String(name)),
 		},
-	}, pulumi.Provider(e.Provider))
+	}
+
+	if ami == "" {
+		ami, err = LatestUbuntuAMI(e, arch)
+		if err != nil {
+			return nil, err
+		}
+		device, err := LatestUbuntuAMIRootDevice(e, arch)
+		if err != nil {
+			return nil, err
+		}
+		rootBlockDevice, err := createNewRootDeviceMapping(device, e.DefaultInstanceStorageSize())
```

Comment:
What's the goal of setting this while the AMI is already `Ubuntu`?
Are you trying to simply change the root disk size? In which case you can simply do:
```
        RootBlockDevice: ec2.InstanceRootBlockDeviceArgs{
            VolumeSize: pulumi.IntPtr(200),
        },
```

Reply:
That is exactly what I needed! Thanks